repo/content/index/info.go

```diff
@@ -26,60 +26,56 @@ type Info interface {
 
 // InfoStruct is an implementation of Info based on a structure.
 type InfoStruct struct {
-	ContentID           ID                   `json:"contentID"`
 	PackBlobID          blob.ID              `json:"packFile,omitempty"`
 	TimestampSeconds    int64                `json:"time"`
 	OriginalLength      uint32               `json:"originalLength"`
 	PackedLength        uint32               `json:"length"`
 	PackOffset          uint32               `json:"packOffset,omitempty"`
+	CompressionHeaderID compression.HeaderID `json:"compression,omitempty"`
+	ContentID           ID                   `json:"contentID"`
 	Deleted             bool                 `json:"deleted"`
 	FormatVersion       byte                 `json:"formatVersion"`
-	CompressionHeaderID compression.HeaderID `json:"compression,omitempty"`
 	EncryptionKeyID     byte                 `json:"encryptionKeyID,omitempty"`
 }
 
 // GetContentID implements the Info interface.
-func (i *InfoStruct) GetContentID() ID { return i.ContentID }
+func (i InfoStruct) GetContentID() ID { return i.ContentID }
```

Comment:
Does changing the type of the receiver from pointer to value make a difference? Why?

Reply:
it should not change any behavior, since we're using InfoStruct in a read-only fashion and the compiler can inline the method and thus avoid copying the structure on stack.